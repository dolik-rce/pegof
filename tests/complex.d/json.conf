input complex.d/json.peg
optimize all
header never
