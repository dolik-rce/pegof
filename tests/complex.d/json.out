%prefix "json"

file <- _ (
    object
    / "[" (
        value ("," value)*
        / _
    ) "]"
) _

object <- "{" (
    _ "\"" (
        "\\\""
        / [^"]
    )* "\"" _ ":" value ("," _ "\"" (
            "\\\""
            / [^"]
        )* "\"" _ ":" value)*
    / _
) "}"

value <- _ (
    object
    / "[" (
        value ("," value)*
        / _
    ) "]"
    / "false"
    / "true"
    / "-"? (
        "0"
        / [1-9] [0-9]*
    ) ("." [0-9]+)? ([Ee] [-+]? [0-9]+)?
    / "\"" (
        "\\\""
        / [^"]
    )* "\""
    / "null"
) _

_ <- [\t\n\r ]*

%%
int main() {
    json_context_t *ctx = json_create(NULL);
    while (json_parse(ctx, NULL));
    json_destroy(ctx);
    return 0;
}
